### PR TITLE
Allow custom style

### LINKED ISSUE #18: Make it possible to pass `style` to header
As Martin suggested in old repo, it would be great to have an ability to define a custom style for a header on a Navigator level.

This is already getting implemented in #1. As a follow-up, we will expose style.

### PR DESCRIPTION
**Pull request by grabbou**
Allows setting custom `style` property on a header to change e.g. backgroundColor. Fixes #18. Also fixes `headerComponent` by fixing an old property reference

**Test plan (required)**

Go to any example in playground and define `header.backgroundColor`. Reload.

### CODE REVIEW

File: src/views/CardStack.js [hunk 1]
```diff
@@ -184,16 +184,17 @@ class CardStack extends React.Component<DefaultProps, Props, void> {
 
   _renderHeader(props: NavigationTransitionProps, headerMode: HeaderMode): ?React.Element<*> {
     const navigation = this._getChildNavigation(props.scene);
-    const header = this.props.router.getScreenConfig(navigation, 'header');
+    const header = this.props.router.getScreenConfig(navigation, 'header') || {};
 
     let HeaderComponent = Header;
-    if (header && header.bar) {
+    if (header.bar) {
```
Comment (satya164) [inline src/views/CardStack.js]: Is the option to change the header component called `header.bar`? It doesn't match what we've in the type definitions https://github.com/react-community/react-navigation/blob/master/src/TypeDefinition.js#L244

Comment (grabbou) [inline src/views/CardStack.js]: Yeah, will send a sep. PR

Comment (grabbou) [inline src/views/CardStack.js]: Actually good note. Good we fix bugs at the same time :)

File: src/views/CardStack.js [hunk 2]
```diff
@@ -184,16 +185,14 @@ class CardStack extends React.Component<DefaultProps, Props, void> {
...
   _renderHeader(props: NavigationTransitionProps, headerMode: HeaderMode): ?React.Element<*> {
     const navigation = this._getChildNavigation(props.scene);
-    const header = this.props.router.getScreenConfig(navigation, 'header');
+    const header = this.props.router.getScreenConfig(navigation, 'header') || {};
 
-    let HeaderComponent = Header;
-    if (header && header.bar) {
-      HeaderComponent = header.bar;
-    }
+    const HeaderComponent = this.props.headerComponent || Header;
```
Comment (satya164) [inline src/views/CardStack.js]: We should use `deafultProps` for this

**Comment by grabbou** [comment]
Since this PR combines two issues, I'll rebase them to two commits that can be rebase and merged.

**Pull request merged**